### PR TITLE
Done payments import per company from xls, xlsx and csv

### PR DESCRIPTION
**Pull request by Ukhanskyi**
[Trello Task](https://trello.com/c/44FtI2m1/42-xls-import-for-payments-per-%D1%81ompany)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [ ] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

Payments import

image

image

image

image

image

### CODE REVIEW

File: app/importers/applications_import.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class ApplicationsImport
+  include ActiveModel::Model
+  require 'roo'
```
Comment (mmarusyk) [inline app/importers/applications_import.rb:3]: It isn't necessary

File: app/controllers/account/admin/payments_imports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Account::Admin::PaymentsImportsController < Account::Admin::AdminController
+  def new
+    @payments_import = PaymentsImport.new
+    respond_to do |format|
```
Comment (mmarusyk) [inline app/controllers/account/admin/payments_imports_controller.rb:4]: Enter

File: app/controllers/account/admin/payments_imports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      format.html { render :new }
+      format.xlsx
+    end
+  end
+
+  def create
+    @payments_import = PaymentsImport.new(params[:payments_import])
+    @payments_import.perform
```
Comment (mmarusyk) [inline app/controllers/account/admin/payments_imports_controller.rb:12]: Enter

File: app/importers/applications_import.rb (hunk 1 above)
Comment (Ukhanskyi) [reply app/importers/applications_import.rb]: Thanks, i removed it

**Review by mmarusyk — DISMISSED** [review]

File: app/controllers/account/admin/payments_imports_controller.rb (hunk 2 above)
Comment (Ukhanskyi) [reply app/controllers/account/admin/payments_imports_controller.rb]: Done

File: app/importers/payments_import.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      row = Hash[[open_spreadsheet.row(1), open_spreadsheet.row(i)].transpose]
+      payment = Payment.new
+      payment.attributes = row_to_hash row, %w[amount date]
+      payment.billing_contract = BillingContract.find_by(contract_num: row["contract_number"])
+      payment
+    end
+  end
+
+  def payments_to_import
+    @payments_to_import ||= load_payments_to_import
```
Comment (mmarusyk) [inline app/importers/payments_import.rb:13]: Do you use @payments_to_import?

Comment (mmarusyk) [inline app/importers/payments_import.rb:13]: Do you use @payments_to_import?

Comment (Ukhanskyi) [reply app/importers/payments_import.rb]: Not now...Thanks

**Review by matsevich — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

**Review by Swooty18 — APPROVED** [review]

**Review by mmarusyk — APPROVED** [review]

File: app/importers/applications_import.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+class ApplicationsImport
+  include ActiveModel::Model
+
+  attr_accessor :file
+
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) } if !attributes.nil? && attributes.key?(:file)
```
Comment (mmarusyk) [inline app/importers/applications_import.rb:7]:
```suggestion
    attributes.each { |name, value| send("#{name}=", value) } if !attributes&.key?(:file)
```

Comment (Ukhanskyi) [reply app/importers/applications_import.rb]: Yes... Your code is more simpler

**Review by IvanDanyliv — APPROVED** [review]

File: app/importers/payments_import.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class PaymentsImport < ApplicationsImport
+  def load_payments_to_import
```
Comment (chu4) [inline app/importers/payments_import.rb:2]: shouldn't they be private?

**Review by chu4 — APPROVED** [review]

**Pull request merged**